dali/operators/reader/video_reader_decoder_gpu_op.cc

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include "dali/operators/reader/video_reader_decoder_gpu_op.h"
+
+#include <string>
+#include <vector>
+
+namespace dali {
+
+VideoReaderDecoderGpu::VideoReaderDecoderGpu(const OpSpec &spec)
+    : DataReader<GPUBackend, VideoSampleGpu>(spec),
+      has_labels_(spec.HasArgument("labels")) {
+      loader_ = InitLoader<VideoLoaderDecoderGpu>(spec);
+}
+
+void VideoReaderDecoderGpu::RunImpl(DeviceWorkspace &ws) {
+  auto &video_output = ws.Output<GPUBackend>(0);
+  int batch_size = GetCurrBatchSize();
+
+  TensorListShape<4> output_shape;
+  output_shape.resize(batch_size);
+
+  for (int sample_id = 0; sample_id < batch_size; ++sample_id) {
+    auto &sample = GetSample(sample_id);
+    output_shape.set_tensor_shape(sample_id, sample.data_.shape());
+  }
+
+  video_output.Resize(output_shape, GetSample(0).data_.type());
```

Comment:
```suggestion
  auto dtype = GetSample(0).data_.type();
  for (int sample_id = 0; sample_id < batch_size; ++sample_id) {
    auto &sample = GetSample(sample_id);
    assert(sample.data_.dtype() == dtype);
    output_shape.set_tensor_shape(sample_id, sample.data_.shape());
  }
  video_output.Resize(output_shape, dtype);
```

Reply:
How about adding SetupImpl just for that?

Reply:
Done